cmd/kubeadm/app/phases/selfhosting/selfhosting.go

```diff
@@ -70,11 +73,17 @@ func CreateSelfHostedControlPlane(client *clientset.Clientset) error {
 
 		// Create the DaemonSet in the API Server
 		if _, err := client.ExtensionsV1beta1().DaemonSets(metav1.NamespaceSystem).Create(ds); err != nil {
-			return fmt.Errorf("failed to create self-hosted %q daemonset [%v]", componentName, err)
+			if !apierrors.IsAlreadyExists(err) {
+				return fmt.Errorf("failed to create self-hosted %q daemonset [%v]", componentName, err)
+			}
+
+			if _, err := client.ExtensionsV1beta1().DaemonSets(metav1.NamespaceSystem).Update(ds); err != nil {
```

Comment:
We may fail on status updates in the DaemonSet so I would make this retriable on 409s. TODO is also fine.